dotnet/src/Connectors/Connectors.AssemblyAI/AudioToText/AssemblyAIAudioToTextService.cs

```diff
@@ -0,0 +1,281 @@
+        if (json.RootElement.TryGetProperty("error", out var property))
+        {
+            throw new KernelException($"Failed to create transcript. Reason: {property.GetString()!}");
+        }
+
+        return json.RootElement.GetProperty("id").GetString()!;
+    }
+
+    private async Task<JsonDocument> WaitForTranscriptToProcessAsync(
+        string transcriptId,
+        PromptExecutionSettings? executionSettings,
+        CancellationToken ct = default
+    )
+    {
+        var url = this.Url($"v2/transcript/{transcriptId}");
+
+        var pollingInterval = TimeSpan.FromMilliseconds(500);
+        if (executionSettings is AssemblyAIAudioToTextExecutionSettings aaiSettings)
+        {
+            pollingInterval = aaiSettings.PollingInterval;
+        }
+
+        while (true)
+        {
+            ct.ThrowIfCancellationRequested();
+            using var request = HttpRequest.CreateGetRequest(url);
+            this.AddDefaultHeaders(request);
+            using var response = await this.SendWithSuccessCheckAsync(this.HttpClient, request, ct).ConfigureAwait(false);
+            using var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+            var json = await JsonDocument.ParseAsync(jsonStream, cancellationToken: ct).ConfigureAwait(false);
+
+            var status = json.RootElement.GetProperty("status").GetString()!;
+            switch (status)
+            {
+                case "processing":
+                case "queued":
+                    await Task.Delay(pollingInterval, ct).ConfigureAwait(false);
+                    break;
+                case "completed":
+                    return json;
+                case "error":
+                    var errorString = json.RootElement.GetProperty("error").GetString()!;
+                    throw new KernelException($"Failed to create transcript. Reason: {errorString}");
+                default:
+                    throw new KernelException("Unexpected transcript status. This code shouldn't be reachable.");
+            }
+        }
+    }
+
+    /// <summary>
+    /// Create a URL string that includes the default BaseUrl if the BaseAddress on HttpClient isn't set.
+    /// </summary>
+    /// <param name="url">URL without base.</param>
+    /// <returns>URL with or without BaseUrl.</returns>
+    private string Url(string url)
+    {
+        return this.HttpClient.BaseAddress is null ? $"{FallbackBaseUrl}{url}" : url;
+    }
+
+    private void AddDefaultHeaders(HttpRequestMessage request)
+    {
+        request.Headers.Authorization = new AuthenticationHeaderValue(this.ApiKey);
+        request.Headers.Add("User-Agent", HttpHeaderConstant.Values.UserAgent);
+        request.Headers.Add(HttpHeaderConstant.Names.SemanticKernelVersion,
+            HttpHeaderConstant.Values.GetAssemblyVersion(typeof(AssemblyAIAudioToTextService)));
+    }
+
+    private async Task<HttpResponseMessage> SendWithSuccessCheckAsync(
```

Comment:
Please use the existing SendWithSuccessCheckAsync utility method and consider parsing the response content, if it really needed, at a higher level to avoid spreading the logic of the SendWithSuccessCheckAsync method throughout the code.

Reply:
I'll remove this for now. We can discuss this in a separate PR.